shell/pages/c/_cluster/_product/projectsnamespaces.vue

```diff
@@ -150,19 +150,9 @@ export default {
       return this.groupPreference === 'none' ? this.rows : this.rowsWithFakeNamespaces;
     },
     rows() {
-      if (this.$store.getters['prefs/get'](DEV)) {
-        return this.activeNamespaces;
-      }
-
-      const isVirtualCluster = this.$store.getters['isVirtualCluster'];
-      const isVirtualProduct = this.$store.getters['currentProduct'].name === HARVESTER;
-
-      return this.activeNamespaces.filter((namespace) => {
-        const isSettingSystemNamespace = this.$store.getters['systemNamespaces'].includes(namespace.metadata.name);
-        const systemNS = namespace.isSystem || namespace.isFleetManaged || isSettingSystemNamespace;
-
-        return isVirtualCluster && isVirtualProduct ? (!systemNS && !namespace.isObscure) : !namespace.isObscure;
-      });
+      // These rows are not filtered in order to stay consistent
+      // with the namespaces available in the top nav of Cluster Explorer.
+      return this.activeNamespaces;

```

Comment:
We still want to filter out system namespaces when `isVirtualCluster` is `true`: all system namespaces are filtered out of `NamespaceFilter` in that scenario.
The `isVirtualProduct` check here is technically redundant but you don't need to fix that here if you don't want to: it'll be fixed/made less harvester-specific when the harvester plugin pr is merged.

Reply:
I changed it to:

```js
rows() {
      const isVirtualCluster = this.$store.getters['isVirtualCluster'];

      if (!isVirtualCluster || this.$store.getters['prefs/get'](DEV)) {
        return this.activeNamespaces;
      }

      // Filter out system namespaces for virtual clusters, e.g. Harvester.
      // Otherwise the unfiltered list should be the same as the list
      // of namespaces available in the top nav of Cluster Explorer.
      return this.activeNamespaces.filter((namespace) => {
        const isSettingSystemNamespace = this.$store.getters['systemNamespaces'].includes(namespace.metadata.name);
        const systemNS = namespace.isSystem || namespace.isFleetManaged || isSettingSystemNamespace;

        return !systemNS && !namespace.isObscure;
      });
    },
```

Reply:
That's perfect, thank you.